tests/50-sd-verify.js

```diff
@@ -23,147 +22,158 @@ const {match} = endpoints.filterByTag({
 
 describe('ecdsa-sd-2023 (verify)', function() {
   describe('ecdsa-sd-2023 (verifiers)', function() {
-    let issuers;
-    let vcHolder;
-    before(async function() {
-      const {match: matchingIssuers} = endpoints.filterByTag({
-        tags: [...tags],
-        property: 'issuers'
-      });
-      const {match: matchingVcHolders} = endpoints.filterByTag({
-        tags: [...holderTags],
-        property: 'vcHolders'
-      });
-      // Uses 'Digital Bazaar' as default issuer to issue a verifiable
-      // credential for the verifier tests.
-      issuers = matchingIssuers.get(issuerName).endpoints;
-      const vcHolders = matchingVcHolders.get(holderName).endpoints;
-      vcHolder = vcHolders[0];
-    });
     // this will tell the report
     // to make an interop matrix with this suite
     this.matrix = true;
     this.report = true;
     this.rowLabel = 'Test Name';
     this.columnLabel = 'Verifier';
     this.implemented = [];
+    const testVectors = {
+      //signedCredentials
+      signed: [],
+      disclosed: {
+        //disclosedCredentials
+        base: [],
+        //nestedDisclosedCredentials
+        nested: [],
+        //disclosedDlCredentialNoIds
+        noIds: [],
+        array: {
+          //disclosedCredentialsWithFullArray
+          full: [],
+          //disclosedCredentialsWithLessThanFullSubArray
+          lessThanFull: [],
+          //disclosedCredentialsWithoutFirstArrayElement
+          missingElements: []
+        }
+      }
+    };
+    before(async function() {
+      const {subjectNestedObjects, subjectHasArrays} = credentials.verify;
+      // create initial signed VCs
+      testVectors.signed = await issueTestData({
+        credential: subjectNestedObjects.document,
+        suite,
+        keyTypes,
+        mandatoryPointers: subjectNestedObjects.mandatoryPointers
+      });
+      const signedVc = testVectors.signed.get(keyTypes[0]);
+      // use initial VCs for a basic selective disclosure test
+      testVectors.disclosed.base = await deriveTestData({
+        selectivePointers: ['/credentialSubject/id'],
+        verifiableCredential: signedVc,
+        keyTypes,
+        suite
+      });
+      // create initial nestedDisclosedCredential from signedVc
+      testVectors.disclosed.nested = await deriveTestData({
+        selectivePointers: subjectNestedObjects.selectivePointers.slice(1, 3),
+        verifiableCredential: signedVc,
+        keyTypes,
+        suite
+      });
+      // copy the first vc
+      const noIdVc = klona(subjectNestedObjects.document);
+      // delete the id
+      delete noIdVc.id;
```

Comment:
Question: why is it important that the ID property is deleted?

Reply:
ids are often used in data tracking so what we're testing here is that it's ok to create a selectively disclosed VC from a VC with out an id. As a later comment mentions the test inputs for sd verify need a once over and some simplification. As they stand the inputs are not that great.